hooks/playbooks/manila_create_default_resources.yml

```diff
@@ -11,3 +11,8 @@
         oc -n {{ namespace }} exec -it pod/openstackclient \
           -- openstack share type create default false \
           --snapshot-support True --create-share-from-snapshot-support True
```

Comment:
Maybe it would make it sense to split it into the create (and not fail if it exists)) and one to assign the additional spec

Reply:
yes; wouldn't it?


```
---
- name: Create Manila resources needed for tempest run
  hosts: "{{ cifmw_target_hook_host | default('localhost') }}"
  gather_facts: false
  vars:
    share_type_name: "default"
    driver_handles_share_servers: "False"
    extra_specs:
      snapshot_support: "True"
      create_share_from_snapshot_support: "True"
  tasks:
    - name: Check if the Manila share type already exists
       environment:
        KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
        PATH: "{{ cifmw_path }}"
      ansible.builtin.shell: |
        oc -n {{ namespace }} exec -it pod/openstackclient \
        -- openstack share type list -f value -c Name | grep -w "{{ share_type_name }}"
      register: share_type_exists
      ignore_errors: true
    - name: Create the Manila share type if it does not exist
      environment:
        KUBECONFIG: "{{ cifmw_openshift_kubeconfig }}"
        PATH: "{{ cifmw_path }}"
      ansible.builtin.shell: |
        oc -n {{ namespace }} exec -it pod/openstackclient \
        openstack share type create "{{ share_type_name }}" \
           "{{ driver_handles_share_servers }}" \
           {{ ' '.join('--extra-specs ' + k + '=' + v for k, v in extra_specs.items()) }}
      when: share_type_exists.rc != 0
```

Reply:
hm this seems as just more verbose 'exists - or create',
it does not cover the case when it may exist but without expected properties

what tosky suggests (and i agree its at least safer) is more like:
- share type create (failed_when: failed && not already-exists)
- share type update to set properties // assuming this does not fail if properties already match (will check)

Reply:
I've updated this to align with both - to do create|ignore-existing + update properties ... and use the syntax/approach based on vars for name and extra_specs dictionary.